lib/benchmark/bench.go

```diff
@@ -175,7 +179,7 @@ func (b *benchmarkThread) execute(measure *benchMeasure) {
 		return
 	}
 	config := b.client.Config
-	client, err := NewClient(&config)
+	client, err := client.NewClient(&config)
 	reader, writer := io.Pipe()
```

Comment:
I would first handle the `err` since if non-nil, anything you do for the client is invalid incl. the client itself.
Also, reader/writer need to be closed when done.